### ISSUE TITLE
csrf.Token() returns an empty token, even though _gorilla_csrf cookie is set

### ISSUE THREAD

**Issue by echlebek** [gorilla/csrf#21]
I'm having trouble getting the Token() helper function to work.

I've wrapped my root-level handler with a csrf.Protect handler, which seems to work. The handler rejects requests that don't specify the correct token, and also sets the _gorilla_csrf cookie on GET requests.

But when I use csrf.Token for setting X-CSRF-Token as described in the docs, I get an empty string.

Looking at the function it seems that an empty string will be returned if the context store does not contain the token secret. But I don't understand how that can be when I'm *also* getting the _gorilla_csrf cookie set to a non-empty value.

Any pointers or clarification would be greatly appreciated.

**Comment by elithrar** [comment]
Hi Eric - can you post your code, or a minimal example that demonstrates
the problem?

**Comment by echlebek** [comment]
I'm trying to write a self-contained example of the problem, will post back later. Thanks!

**Comment by elithrar** [comment]
Remember that you need to set `csrf.Secure(false)` during development.

**Comment by echlebek** [comment]
Huge facepalm moment here.

In package main, where I initialize the root handler with csrf.Protect, I was importing from github.com/gorilla/csrf. However in my handlers, which exist in another package, I was importing a vendored version of the library. This happened without me noticing it via goimports.

I'm not entirely sure why it didn't just work, perhaps different code versions, or some package-level state, but I think it's good that the library refused to work properly in this situation.

Sorry for filing a spurious issue!

**Comment by elithrar** [comment]
No problems. csrf.Token(r) has existed since day one, so not sure why the
vendored version would have caused issues. Still, glad to hear that it's
resolved!

**Comment by f03lipe** [comment]
I seem to be experiencing the exact same thing. :weary: 
Can you share your solution to this, @echlebek?

**Comment by elithrar** [comment]
@f03lipe Can you show your code and make sure you're on the latest version of the library?

**Comment by f03lipe** [comment]
@elithrar thank you for the reply. The context is this: I have the following directory structure:
```
server.go
handlers/
   main.go
```
Both import `csrf`. They were working as expected when I was developing. Then I used `godep save -r`, to save dependencies in the `Godeps/_workspace` dir. Because of the `-r`, the `server.go` import was changed `[path to app]Godeps/_workspace/src/github.com/gorilla/csrf`; but the import in `handlers/main.go` remained unchanged (still trying to figure out why `godep` behaves like this).

Now, when I run the server, `csrf.Token()` returns an empty token. Is this a godep bug?
